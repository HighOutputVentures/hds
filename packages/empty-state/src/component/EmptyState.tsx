import { AddIcon } from '@chakra-ui/icons';
import { Box, Text, Flex, Button } from '@chakra-ui/react';
import React, { FC, ReactElement } from 'react';

export interface EmptyStateProps {
  header: string;
  subHeader: string;
  icon?: ReactElement;
  primaryLabel: string;
  secondaryLabel: string;
  onClear?: () => void;
  onSubmit?: () => void;
}

const EmptyState: FC<EmptyStateProps> = ({
  header,
  subHeader,
  icon,
  primaryLabel,
  secondaryLabel,
  onClear,
  onSubmit,
}) => (
  <Box>
    <Flex
      direction="column"
      w={{ base: 'auto', lg: '48rem' }}
      justifyContent="center"
      p="24px"
      textAlign="center"
      alignItems="center"
    >
      {icon && (
        <Box my="1rem" bgColor="">
          {icon}
        </Box>
      )}
      <Text fontWeight={500} size={{ base: 'label-sm-default', lg: 'label-md-default' }}>
        {header}
      </Text>
      <Text
        w={{ lg: 352 }}
        color="gray.500"
        size={{ base: 'label-sm-default', lg: 'paragraphs-md-default' }}
        my="0.5rem"
      >
        {subHeader}
      </Text>

      <Flex gap={4} mt="1rem">
        {secondaryLabel && (
          <Button onClick={onClear} variant="outline">
            {secondaryLabel}
          </Button>
        )}
        {primaryLabel && (
          <Button leftIcon={<AddIcon />} onClick={onSubmit}>
            {primaryLabel}
          </Button>
        )}
      </Flex>
    </Flex>
  </Box>
);

export default EmptyState;
