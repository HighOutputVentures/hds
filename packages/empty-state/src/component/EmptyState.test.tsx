import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import React from 'react';
import EmptyState from './EmptyState';

describe('EmptyState', () => {
  it('should render EmptyState', () => {
    const { container } = render(
      <EmptyState
        header={`No projects found`}
        subHeader="Your search “Landing page design” did not match any projects. Please try again."
        primaryLabel="Changelog"
        secondaryLabel="Dismiss"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render EmptyState', () => {
    const { getByText } = render(
      <EmptyState
        header={`No projects found`}
        subHeader="Your search “Landing page design” did not match any projects. Please try again."
        primaryLabel="Changelog"
        secondaryLabel="Dismiss"
      />,
    );

    expect(getByText(`No projects found`)).toBeInTheDocument();
    expect(
      getByText(
        `Your search “Landing page design” did not match any projects. Please try again.`,
      ),
    ).toBeInTheDocument();
  });
});
