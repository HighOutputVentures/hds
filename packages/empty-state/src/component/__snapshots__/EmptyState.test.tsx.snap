// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render EmptyState 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-km9j2t"
    >
      <p
        class="chakra-text css-10rvbm3"
      >
        No projects found
      </p>
      <p
        class="chakra-text css-17jjy1e"
      >
        Your search “Landing page design” did not match any projects. Please try again.
      </p>
      <div
        class="css-8vgms9"
      >
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          Dismiss
        </button>
        <button
          class="chakra-button css-4xx2wk"
          type="button"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden="true"
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </span>
          Changelog
        </button>
      </div>
    </div>
  </div>
</div>
`;
